staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -199,7 +193,18 @@ func (s *Scheme) AddKnownTypeWithName(gvk schema.GroupVersionKind, obj Object) {
 		panic("All types must be pointers to structs.")
 	}
 
+	if oldT, found := s.gvkToType[gvk]; found && oldT != t {
```

Comment:
we're already panicing on invalid registrations... why not just panic in this case as well? plumbing/preserving stack traces seems really weird